api/scripts/daily-digest.ts

```diff
@@ -0,0 +1,125 @@
+        'UserId': user.id
+      }});
+
+      groups.forEach(async group => {
+        const recordingData = {};
+        const name = await models.Group.findOne({where: {
+          'id': group.GroupId
+        }});
+
+        const stations = await models.Station.findAll({where: {
+          'GroupId': group.GroupId
+        }});
+        const stationsArray = [];
+        stations.forEach(async station => {
+          stationsArray.push(station.id);
+        });
+
+        const oneWeekAgo = new Date();
+        const now = new Date();
+        oneWeekAgo.setHours(oneWeekAgo.getHours() - 168);
+
+        const params: MonitoringParams = {
+          stations: stationsArray,
+          groups: [group.GroupId],
+          page: 1,
+          pageSize: 100,
+          from: oneWeekAgo,
+          until: now,
+          types: [RecordingType.ThermalRaw]
+        };
+
+        const viewAsSuperAdmin = true;
+        const searchDetails = await calculateMonitoringPageCriteria(
+          requestUser,
+          params,
+          viewAsSuperAdmin
+        );
+      
+        searchDetails.compareAi = "Master";
+        searchDetails.types = params.types;
+      
+        const visits = await generateVisits(
+          requestUser.id,
+          searchDetails,
+          viewAsSuperAdmin
+        );
+        
+        (visits as unknown as Array<any>).forEach(async visit => {
+          if (recordingData[visit.classification] !== undefined) {
+            recordingData[visit.classification] += 1;
+          } else {
+            recordingData[visit.classification] = 1;
+          }
+        });
+
+        const visitsTotal = (visits as unknown as Array<any>).length;
+        const speciesListArray = Object.entries(recordingData).map(([species, count]) => {
+          return { species, count, widthPercent: 100 / Object.keys(recordingData).length };
+        });
+
+        const interpolants: {
+          [key: string]: any;
+        } = {
+          groupName: `${name.groupName}`,
+          groupURL: `https://browse-next.cacophony.org.nz/${name.groupName}`,
+          visitsTotal: visitsTotal,
+          speciesList: speciesListArray,
+          recordingUrl: "https://browse-next.cacophony.org.nz/",
+          emailSettingsUrl: "https://browse-next.cacophony.org.nz/",
+          cacophonyBrowseUrl: "https://browse-next.cacophony.org.nz/",
+          cacophonyDisplayUrl: "Cacophony monitoring platform",
+        };
+
+        const { text, html } = await createEmailWithTemplate(
+          templateFilename,
+          interpolants
+        );
+        
+        const emailData = {
+          text: text,
+          from: config.smtpDetails.from_name,
+          to: `${user.userName} <>`,
+          subject: "Daily digest",
+          attachment: [{ data: html.replace('<div id="speciesListContainer"', `<div id="speciesListContainer"`), alternative: true }],
+        };
+
+        client.send(emailData, (err, message) => {
```

Comment:
You seem to be sending a separate digest email for each group a user is a member of.  I think it would be better to have the digest email be a single email breaks down the activity per group.